### PR TITLE
ref(issue-details): Replace App only with Relevant only

### PR DESCRIPTION
**Pull request by priscilawebdev**
INGEST-74

**Preview:**

image

mobile:

image

### CODE REVIEW

File: static/app/types/index.tsx [hunk 1]
```diff
@@ -1978,11 +1978,22 @@ export type Frame = {
   symbolicatorStatus: SymbolicatorStatus;
   trust: any | null;
   vars: Record<string, any> | null;
+  addrMode?: string;
```
Comment (untitaker) [inline static/app/types/index.tsx:1981]: I think this fits well with instructionAddr, it's topically related

File: static/app/types/index.tsx [hunk 2]
```diff
@@ -1978,11 +1978,22 @@ export type Frame = {
...
+  isPrefix?: boolean;
+  minGroupingLevel?: number;
 };
 
+export enum FrameCategory {
+  SENTINEL = 'sentinel',
+  PREFIX = 'prefix',
+  GROUPING = 'grouping',
+  IN_APP = 'in_app',
+}
```
Comment (untitaker) [inline static/app/types/index.tsx:1995]: a frame can have all of those at once

also category is already a used term in the backend, would not recommend using that term here

File: static/app/views/organizationGroupDetails/grouping/newIssue.tsx [hunk 3]
```diff
@@ -6,21 +6,28 @@ import TimeSince from 'app/components/timeSince';
...
       <EventDetails>
         <EventOrGroupHeader
-          data={sampleEvent}
+          data={{
+            ...sampleEvent,
+            metadata: {
+              ...sampleEvent.metadata,
+              current_level: group?.metadata?.current_level,
+            },
+          }}
```
Comment (untitaker) [inline static/app/views/organizationGroupDetails/grouping/newIssue.tsx:30]: in `grouping/grouping.tsx` we already do a hack like this when creating `NewIssue`, maybe we can construct the event in one go

File: static/app/components/events/interfaces/frame/category.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import {t} from 'app/locale';
+import {FrameCategory} from 'app/types';
+import {Theme} from 'app/utils/theme';
+
+type Props = {
+  theme: Theme;
+  category: FrameCategory;
+};
+
+function Category({category, theme}: Props) {
```
Comment (untitaker) [inline static/app/components/events/interfaces/frame/category.tsx:15]: as per other comment, I would definetly not call this category. call it `flag`, `label` or anything else... frames can have multiple labels/flags, but the category is something like "ui"/"driver"/etc, if we ever expose this via frontend we will have naming collisions

File: static/app/components/eventTitleTreeLabel.tsx [hunk 5]
```diff
@@ -22,15 +22,15 @@ function EventTitleTreeLabel({treeLabel}: Props) {
           if (index !== firstFourParts.length - 1) {
             return (
               <Fragment key={index}>
-                <PriorityPart highlight={highlight}>{label}</PriorityPart>
```
Comment (untitaker) [inline static/app/components/eventTitleTreeLabel.tsx:25]: this is an unrelated change entirely, right? can you describe what you're fixing here (wrt the css changes, not this line specifically)

File: static/app/components/events/interfaces/crashContent/index.tsx [hunk 6]
```diff
@@ -10,6 +10,8 @@ type Props = Pick<
   ExceptionProps,
   'stackType' | 'stackView' | 'projectId' | 'event' | 'newestFirst'
 > & {
+  hasGroupingTreeUI: boolean;
+  groupingCurrentLevel?: Group['metadata']['current_level'];
```
Comment (untitaker) [inline static/app/components/events/interfaces/crashContent/index.tsx:14]: isn't this already in ExceptionProps? you could pick them out of there

File: static/app/components/events/interfaces/crashHeader/crashActions.tsx [hunk 7]
```diff
@@ -72,15 +74,24 @@ const CrashActions = ({
...
-          <Button
-            barId={STACK_VIEW.APP}
-            size="xsmall"
-            onClick={setStackView(STACK_VIEW.APP)}
-          >
-            {t('App Only')}
-          </Button>
-        )}
+        {hasSystemFrames &&
+          (hasGroupingTreeUI ? (
```
Comment (untitaker) [inline static/app/components/events/interfaces/crashHeader/crashActions.tsx:78]: can't this conditional be just around the `t(..)`?

`hasGroupingTreeUI ? t('Relevant Only') : t('App Only')`

File: static/app/components/events/interfaces/exception.tsx [hunk 8]
```diff
@@ -1,97 +1,105 @@
...
+  // thread interface.
+  if (eventHasThreads) {
+    return null;
+  }
 
-  render() {
-    const eventHasThreads = !!this.props.event.entries.find(
-      entry => entry.type === 'threads'
-    );
+  function handleChange({
```
Comment (untitaker) [inline static/app/components/events/interfaces/exception.tsx:47]: noob question, won't this function be created on every render? https://stackoverflow.com/a/56459512/1544347

it appears you have a lot of nested helper functions like that, I wonder if that affects performance for large stacktraces

in either case the refactor to class-based components makes this diff hard to read

File: static/app/components/events/interfaces/exceptionStacktraceContent.tsx [hunk 9]
```diff
@@ -61,6 +67,27 @@ const ExceptionStacktraceContent = ({
    * It is easier to fix the UI logic to show a non-empty stack trace for chained exceptions
    */
 
+  const stackTracePlatform =
+    ((data?.frames ?? []).find(frame => frame.platform) || platform) ?? 'other';
+
+  if (
+    hasGroupingTreeUI &&
+    (stackTracePlatform === 'native' || stackTracePlatform === 'cocoa')
```
Comment (untitaker) [inline static/app/components/events/interfaces/exceptionStacktraceContent.tsx:75]: I'm wondering if it would be better to show StacktraceContentV2 for all platforms even if the other platforms haven't been implemented yet. After all one has the labels and the other one doesn't.

**Review by untitaker — CHANGES_REQUESTED** [review]
this diff is really a lot, it mixes stacktrace component refactor with function/class conversion with the actual feature development and a bugfix for group title

I really wish this PR would be split up, it's just really hard to review like this

File: static/app/components/eventTitleTreeLabel.tsx (hunk 5 above)
Comment (untitaker) [reply static/app/components/eventTitleTreeLabel.tsx:25]: it appears to be a bug in max grouping level where the last elements linewrap/overflow if there are too many treelabelparts

ref https://getsentry.atlassian.net/browse/INGEST-172

File: static/app/types/index.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/types/index.tsx:1981]: this prop was just moved to the bottom of the object since it's optional

**Comment by priscilawebdev** [comment]
Thanks for the feedback @untitaker... yes, I was planning to convert this draft PR into multiple PRs because now it's really hard to review

File: static/app/components/eventTitleTreeLabel.tsx (hunk 5 above)
Comment (priscilawebdev) [reply static/app/components/eventTitleTreeLabel.tsx]: I removed this bug fix from this PR ;-)

File: static/app/components/events/interfaces/exceptionStacktraceContent.tsx (hunk 9 above)
Comment (priscilawebdev) [reply static/app/components/events/interfaces/exceptionStacktraceContent.tsx:75]: I think I would prefer to leave it as it is until I check what the UI looks like for other platforms

File: static/app/components/events/interfaces/exception.tsx (hunk 8 above)
Comment (priscilawebdev) [reply static/app/components/events/interfaces/exception.tsx:47]: yes, it will. "does it impact performance in a significant way?" it depends on the case, but I would say that in most of the cases no. We can use some performance optimization functions if we notice that the performance of large stack traces is bad

**Review by matejminar — APPROVED** [review]
With Markus' comments looks good to me.
This is still feature flagged, right?

- Should we test this on staging?
- Does this affect stacktrace hover preview?

**Comment by priscilawebdev** [comment]
@matejminar yes it's feature flagged 👍

File: static/app/components/events/interfaces/frame/line.tsx [hunk 10]
```diff
@@ -301,11 +322,66 @@ export class Line extends React.Component<Props, State> {
...
+                isHoverPreviewed={isHoverPreviewed}
+              />
+            )}
+            <StyledSymbol
+              frame={data}
+              showCompleteFunctionName={!!showCompleteFunctionName}
+              onFunctionNameToggle={onFunctionNameToggle}
+              isHoverPreviewed={isHoverPreviewed}
+            />
+            <FrameCategories>
```
Comment (untitaker) [inline static/app/components/events/interfaces/frame/line.tsx:375]:
```suggestion
            <FrameGroupingTags>
```

**Review by untitaker — APPROVED** [review]
